app/Main.hs

```diff
@@ -75,10 +75,14 @@ initDbAndFetcher env Config {..} =
             latestBlock' <- getLastBlock
             let firstBlock =
                   if useLatest
-                    then fromMaybe blockInfo latestBlock'
+                    then maybe blockInfo (uncurry BlockInfo) latestBlock'
                     else blockInfo
-            initLastBlock firstBlock
-            updateLastBlock firstBlock
+            case firstBlock of
+              (BlockInfo slot hash) -> do
+                initLastBlock slot hash
+                updateLastBlock slot hash
+              BlockOrigin ->
+                error ""
```

Comment:
Empty error

Reply:
I assume this might happen with a node that doesnt start syncing?